server/src/test/java/org/elasticsearch/index/mapper/ObjectMapperMergeTests.java

```diff
@@ -93,7 +93,16 @@ public void testMergeWhenEnablingField() {
         // WHEN merging mappings
         // THEN a MapperException is thrown with an excepted message
         MapperException e = expectThrows(MapperException.class, () -> rootObjectMapper.merge(mergeWith));
-        assertEquals("Can't update attribute for type [type1.disabled.enabled] in index mapping", e.getMessage());
+        assertEquals("The [enabled] parameter can't be updated for the object mapping [disabled].", e.getMessage());
+    }
+
+    public void testDisableRootMapper() {
+        String type = MapperService.SINGLE_MAPPING_NAME;
+        ObjectMapper firstMapper = createRootObjectMapper(type, true, Collections.emptyMap());
+        ObjectMapper secondMapper = createRootObjectMapper(type, false, Collections.emptyMap());
+
+        MapperException e = expectThrows(MapperException.class, () -> firstMapper.merge(secondMapper));
+        assertEquals("The [enabled] parameter can't be updated for the object mapping [" + type + "].", e.getMessage());
```

Comment:
It's pretty confusing that this error message will contain the singleton type name: "The [enabled] parameter can't be updated for the object mapping [_doc]." I think we should address this in a separate, more general change to not pass in the type as the root object mapper's name.